pkg/actors/actors.go

```diff
@@ -718,6 +720,32 @@ func (a *actorsRuntime) GetState(ctx context.Context, req *GetStateRequest) (*St
 	}, nil
 }
 
+func (a *actorsRuntime) DeleteState(ctx context.Context, req *DeleteStateRequest) (*runtimev1pb.DeleteActorStateResponse, error) {
+	storeName, store, err := a.stateStore()
+	if err != nil {
+		return &runtimev1pb.DeleteActorStateResponse{}, err
+	}
+
+	key := constructCompositeKey(a.actorsConfig.Config.AppID, req.ActorID, req.ActorType, req.ActorKey())
+
+	policyRunner := resiliency.NewRunner[state.DeleteWithPrefixResponse](ctx,
+		a.resiliency.ComponentOutboundPolicy(storeName, resiliency.Statestore),
+	)
+
+	storeReq := state.DeleteWithPrefixRequest{
+		Prefix: key,
+	}
+
+	prefixDelete := store.(state.DeleteWithPrefix)
```

Comment:
Need to check if this type casting is possible. If the state store doesn't support DeleteWithPrefix, this will panic.


```suggestion
	prefixDelete, ok := store.(state.DeleteWithPrefix)
```

Then check if `!ok` return an error that says that the operation isn't possible.